dallinger/experiment.py

```diff
@@ -1307,3 +1334,34 @@ def load():
     except ImportError:
         logger.error("Could not import experiment.")
         raise
+
+
+EXPERIMENT_ROUTE_REGISTRATIONS = []
+
+
+def experiment_route(rule, *args, **kwargs):
+    """Creates a decorator to register experiment functions or classmethods as
+    routes on the ``Experiment.experiment_routes`` blueprint. Accepts all
+    standard flask ``route`` arguments. The registration is deferred until
+    experiment server setup to allow routes to be overridden.
+
+    :returns: A decorator to register methods from a class as experiment routes.
+    """
+    registered_routes = EXPERIMENT_ROUTE_REGISTRATIONS
+    route = {
+        "rule": rule,
+        "args": args,
+        "kwargs": tuple(kwargs.items()),
+    }
+
+    def new_func(func):
```

Comment:
I've refactored out the shared function, it's a little weird because it's a sort of meta-meta-decorator, but it avoids all the duplicated code I think.